store/verkle/proof.go

```diff
@@ -73,29 +84,32 @@ func (p *ProofOp) Run(args [][]byte) ([][]byte, error) {
 }
 
 func (p *ProofOp) GetKey() []byte {
-	return p.Key
+	return p.Proof.Key
 }
 
-func (p *ProofOp) ProofOp() tmmerkle.ProofOp {
+func (p *ProofOp) ProofOp() crypto.ProofOp {
 	var data bytes.Buffer
 	enc := gob.NewEncoder(&data)
-	err := enc.Encode(p)
+	err := enc.Encode(p.Proof)
```

Comment:
Do we really want to encode using `gob` ? There are better enc/dec schemes + are we sure we will only open this under `go` or `C`?